ouroboros-consensus/src/Ouroboros/Consensus/Mempool/Impl.hs

```diff
@@ -366,12 +368,24 @@ implAddTxs mpEnv accum txs = assert (all txInvariant txs) $ do
       -- (those not yet validated due to the mempool capacity being reached).
     validateNewUntilMempoolFull []      vr = pure (vr, [])
     validateNewUntilMempoolFull (t:ts)  vr = do
-      mempoolSize <- getMempoolSize mpEnv
+      -- Get the current mempool size
+      mpsz@MempoolSize { msNumBytes } <- getMempoolSize mpEnv
+
+      -- Determine what the mempool size would be if we were to commit the new
+      -- transactions we've validated thus far and also the next transaction
+      -- to validate, 't'.
+      -- If this value is greater than the 'MempoolCapacity', then we know not
+      -- to continue validating at this time.
+      let MempoolSize _ nextMsNumBytes =
+               mpsz
+            <> txsToMempoolSize (vrNewValid vr)
+            <> txsToMempoolSize [t]
```

Comment:
Hmm, my suggestion was not that good after all. What about the following:
```haskell
let newTxsBytes    = sum (txSize <$> vrNewValid vr) + txSize t
    newSizeInBytes = curSizeInBytes + newTxsBytes
```

Reply:
I agree, I was wondering whether I was the only one that found it a bit unclear.